data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -193,6 +195,7 @@ struct TraceExporterWorkers {
 #[derive(Debug)]
 pub struct TraceExporter {
     endpoint: Endpoint,
+    info_endpoint: Arc<Endpoint>,
```

Comment:
It feels a little bit odd that `TraceExporter` has to be aware of the info endpoint beyond creating the `AgentInfoFetcher` in the builder and has to be responsible for calling `fetch_info_with_state`. 

Ideally, I think an event bus concept where the AgentInfoFetcher can be notified of a response header change would be a good idea. I don't think that's prudent as part of this PR. But what about something like a `ResponseObserver` that lives in agent_info and gets created at build time along with the AgentInfoFetcher (you can add a factory in agent_info that builds both at the same time)?